diff -ur vendor_orig/magento/module-catalog/Model/ResourceModel/Product/BaseSelectProcessorInterface.php vendor/magento/module-catalog/Model/ResourceModel/Product/BaseSelectProcessorInterface.php
--- vendor_orig/magento/module-catalog/Model/ResourceModel/Product/BaseSelectProcessorInterface.php	2018-07-28 12:17:02.000000000 +0100
+++ vendor/magento/module-catalog/Model/ResourceModel/Product/BaseSelectProcessorInterface.php	2018-11-28 15:25:56.000000000 +0000
@@ -1,6 +1,6 @@
 <?php
 /**
- * Copyright © 2013-2017 Magento, Inc. All rights reserved.
+ * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
 namespace Magento\Catalog\Model\ResourceModel\Product;
@@ -9,6 +9,8 @@
 
 /**
  * Interface BaseSelectProcessorInterface
+ * @api
+ * @since 101.0.3
  */
 interface BaseSelectProcessorInterface
 {
@@ -20,6 +22,7 @@
     /**
      * @param Select $select
      * @return Select
+     * @since 101.0.3
      */
     public function process(Select $select);
 }
