diff -ur -N vendor_orig/magento/framework/App/Request/ValidatorInterface.php vendor/magento/framework/App/Request/ValidatorInterface.php
--- vendor_orig/magento/framework/App/Request/ValidatorInterface.php    2020-04-13 17:35:44.000000000 +0000
+++ vendor/magento/framework/App/Request/ValidatorInterface.php 2022-09-12 14:47:34.000000000 +0000
@@ -3,7 +3,6 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
-
 declare(strict_types=1);

 namespace Magento\Framework\App\Request;
@@ -13,6 +12,8 @@

 /**
  * Validate interface before giving passing it to an ActionInterface.
+ *
+ * @api
  */
 interface ValidatorInterface
 {