diff -ur -N vendor_orig/magento/module-weee/Model/Total/Quote/Weee.php vendor/magento/module-weee/Model/Total/Quote/Weee.php
--- vendor_orig/magento/module-weee/Model/Total/Quote/Weee.php	2023-08-04 10:41:06
+++ vendor/magento/module-weee/Model/Total/Quote/Weee.php	2023-08-04 10:41:38
@@ -3,7 +3,7 @@
 
 class Weee
 {
-    public function __construct()
+    public function __construct($someNewParam = 123)
     {
         // do stuff
     }
